control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -128,6 +152,47 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, err
 	}
 
+	// Return early if deleted
+	if !hostedControlPlane.DeletionTimestamp.IsZero() {
+		if err := r.delete(ctx, hostedControlPlane); err != nil {
+			r.Log.Error(err, "failed to delete cluster")
+			return ctrl.Result{}, err
+		}
+		if controllerutil.ContainsFinalizer(hostedControlPlane, finalizer) {
+			controllerutil.RemoveFinalizer(hostedControlPlane, finalizer)
+			if err := r.Update(ctx, hostedControlPlane); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to remove finalizer from cluster: %w", err)
+			}
+		}
+		return ctrl.Result{}, nil
+	}
+
+	patchHelper, err := patch.NewHelper(hostedControlPlane, r.Client)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to init patch helper: %w", err)
+	}
+
+	oldStatus := hostedControlPlane.Status.DeepCopy()
+
+	// Sync any status changes during the reconciliation
+	defer func() {
+		availableCondition := getAvailableCondition(hostedControlPlane)
+
+		if len(oldStatus.Conditions) != 0 {
+			oldAvailableCondition := &oldStatus.Conditions[0]
+			if reflect.DeepEqual(oldAvailableCondition, availableCondition) {
+				// No change to status
+				return
+			}
+		}
+		availableCondition.LastTransitionTime = metav1.Now()
+
+		if err := patchHelper.Patch(ctx, hostedControlPlane); err != nil {
```

Comment:
Should `Client.Status().Update()` be used?

Reply:
`patchHelper` was what was used before so I just kept it.  however, it creates a code dep on CAPI.  i'm in favor of changing it but wanted to keep this focused on the status change.